nym-vpnd/src/command_interface/credential_import.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use maplit::hashmap;
+use nym_vpn_proto::{import_error::ImportErrorType, ImportError as ProtoImportError};
+
+use crate::service::ImportCredentialError;
+
+impl From<ImportCredentialError> for ProtoImportError {
+    fn from(err: ImportCredentialError) -> Self {
+        match err {
+            ImportCredentialError::VpnRunning => ProtoImportError {
+                kind: ImportErrorType::VpnRunning as i32,
+                message: err.to_string(),
+                details: Default::default(),
+            },
+            ImportCredentialError::CredentialAlreadyImported => ProtoImportError {
+                kind: ImportErrorType::CredentialAlreadyImported as i32,
+                message: err.to_string(),
+                details: Default::default(),
+            },
+            ImportCredentialError::StorageError(_) => ProtoImportError {
+                kind: ImportErrorType::StorageError as i32,
+                message: err.to_string(),
+                details: Default::default(),
+            },
+            ImportCredentialError::Generic(_) => ProtoImportError {
+                kind: ImportErrorType::Generic as i32,
+                message: err.to_string(),
+                details: Default::default(),
+            },
+            ImportCredentialError::DeserializationFailure {
+                reason: _,
+                ref location,
+            } => ProtoImportError {
+                kind: ImportErrorType::DeserializationFailure as i32,
+                message: err.to_string(),
+                details: hashmap! { "location".to_string() => location.to_string_lossy().to_string() },
```

Comment:
Do we actually need a separate crate and dep for this macro? maybe we could have a helper function like `fn into_map(k: impl<Into<String>>, v: impl<Into<String>) -> HashMap<String, String>`, then you could remove those `to_string()` and instead call it via `details: into_map("location", location)`; alternatively, but that's just thinking aloud because it would be an overkill: some trait + macro to automatically pull the field name


Also I'm not entirely convinced by storing structured data in an unstructured hashmap but I guess that's the limitation of protobuf for us...

Reply:
So yeah, this was my initial approach, but the complexity started to pile up once I wanted to have multiple entries, `into_map([(k1,v1),(k2,v2)])`, where v1 and v2 are different types (but both implement to_string)